src/Features/Core/Portable/Microsoft.CodeAnalysis.Features.csproj

```diff
@@ -127,6 +127,7 @@
   <ItemGroup>
     <PackageReference Include="System.Threading.Tasks.Extensions" Version="$(SystemThreadingTasksExtensionsVersion)" />
     <PackageReference Include="Microsoft.DiaSymReader" Version="$(MicrosoftDiaSymReaderVersion)" />
+    <PackageReference Include="Microsoft.CodeAnalysis.FlowAnalysis.Utilities" Version="$(MicrosoftCodeAnalysisFlowAnalysisUtilitiesVersion)" PrivateAssets="all" />
```

Comment:
:exclamation: this dependency addition will need a design review

Reply:
Certainly. Few things to note:
1. The assembly will not be loaded on startup or even on opening any C#/VB project. It will be lazily loaded only when we see a first code block which has disposable creation/field.
2. It is strong name signed, so we shouldn't have to worry about version conflicts from different version being loaded by future FxCop analyzer NuGet packages.
3. We are not taking an API dependency here, but purely a functionality dependency on an internal framework that we want to share between analyzers in two repos that we own.
4. Ported analyzers added in this PR are by themselves pretty trivial. The underlying dispose dataflow analysis is the non-trivial shared part. I would recommend reviewing the added unit tests for understanding the analysis semantics.

I did discuss this dependency with @jinujoseph and @vatsalyaagrawal before working on this PR. However, I agree we should discuss and confirm this in a design meeting.

Reply:
@sharwell - I had offline discussion with @vatsalyaagrawal @jinujoseph and @heejaechang and given that we plan to add more (non-analyzer) IDE features based on dataflow analysis, taking the NuGet approach is much more preferable over putting these two dispose analyzers and the FlowAnalysis Utilities assemblies into a VSIX generated from analyzers repo. I will stick to the current approach in this PR.

Reply:
>  It will be lazily loaded only when we see a first code block which has disposable creation/field.

Looking at the source, it seems we are using `DisposeAnalysisHelper` when registering compilation start action, wouldn't it cause the new assembly load? I'm not saying we shouldn't do this, just curious.

Reply:
Yes, that is correct - that was my oversight. If we see issues in loading this assembly upfront, I can make it lazily computed.